client/tx/tx.go

```diff
@@ -310,6 +311,46 @@ func PrepareFactory(clientCtx client.Context, txf Factory) (Factory, error) {
 	return txf, nil
 }
 
+// SignWithPrivKey signs a given tx with the given private key, and returns the
+// corresponding SignatureV2 if the signing is successful.
+func SignWithPrivKey(
+	signMode signing.SignMode,
+	signerData authsigning.SignerData,
+	txBuilder client.TxBuilder,
+	priv crypto.PrivKey,
```

Comment:
Remove the other function, we're left with this `SignWithPrivKey` one, with hopefully improved readability.